File: Daily Test/C/Merge String Values - Without Shuffling 7.10.2025.c
#include <stdio.h>
#include <string.h>

int main() {
    char s1[110], s2[110], result[220];
    scanf("%s%s", s1, s2);
    int i = 0, j = 0, k = 0;
    int n1 = strlen(s1), n2 = strlen(s2);
    while (i < n1 || j < n2) {
        if (i < n1 && j < n2 && s1[i] == s2[j]) {
            char ch = s1[i];
            while (i < n1 && s1[i] == ch) result[k++] = s1[i++];
            while (j < n2 && s2[j] == ch) result[k++] = s2[j++];
        } else {
            if (i < n1) result[k++] = s1[i++];
            else if (j < n2) result[k++] = s2[j++];
        }
    }
    result[k] = '';
    printf("%s
", result);
    return 0;
}

File: Daily Test/Python/Merge String Values - Without Shuffling 7.10.2025.py
s1 = input().strip()
s2 = input().strip()

i = 0
j = 0
result = ""

while i < len(s1) or j < len(s2):
    if i < len(s1) and j < len(s2) and s1[i] == s2[j]:
        ch = s1[i]
        while i < len(s1) and s1[i] == ch:
            result += s1[i]
            i += 1
        while j < len(s2) and s2[j] == ch:
            result += s2[j]
            j += 1
    else:
        if i < len(s1):
            result += s1[i]
            i += 1
        elif j < len(s2):
            result += s2[j]
            j += 1
print(result)